pubsub/system-test/subscriptions.test.js

```diff
@@ -85,117 +73,111 @@ describe(`pubsub:subscriptions`, () => {
     // Listing is eventually consistent. Give the indexes time to update.
     setTimeout(() => {
       const output = run(`${cmd} list`, cwd);
-      assert.notEqual(output.indexOf(`Subscriptions:`), -1);
-      assert.notEqual(output.indexOf(fullSubscriptionNameOne), -1);
-      assert.notEqual(output.indexOf(fullSubscriptionNameTwo), -1);
+      assert.equal(output.includes(`Subscriptions:`), true);
+      assert.equal(output.includes(fullSubscriptionNameOne), true);
+      assert.equal(output.includes(fullSubscriptionNameTwo), true);
       done();
     }, 5000);
   });
 
-  it(`should list subscriptions for a topic`, (done) => {
-    // Listing is eventually consistent. Give the indexes time to update.
+  it(`should list subscriptions for a topic`, () => {
     const output = run(`${cmd} list ${topicName}`, cwd);
-    assert.notEqual(output.indexOf(`Subscriptions for ${topicName}:`), -1);
-    assert.notEqual(output.indexOf(fullSubscriptionNameOne), -1);
-    assert.notEqual(output.indexOf(fullSubscriptionNameTwo), -1);
-    done();
+    assert.equal(output.includes(`Subscriptions for ${topicName}:`), true);
+    assert.equal(output.includes(fullSubscriptionNameOne), true);
+    assert.equal(output.includes(fullSubscriptionNameTwo), true);
   });
 
-  it(`should pull messages`, (done) => {
+  it(`should pull messages`, () => {
     const expected = `Hello, world!`;
-    pubsub.topic(topicName).publish({ data: expected }, (err, messageIds) => {
-      assert.ifError(err);
-      setTimeout(() => {
+    return pubsub.topic(topicName).publish(expected)
+      .then((results) => {
+        const messageIds = results[0];
         const output = run(`${cmd} pull ${subscriptionNameOne}`, cwd);
         const expectedOutput = `Received ${messageIds.length} messages.\n` +
```

Comment:
At that point we're testing `google-cloud-node`, which already has thorough tests of its own.